kong/plugins/datadog/handler.lua

```diff
@@ -105,11 +117,32 @@ function DatadogHandler:log(conf)
     return
   end
 
-  local message = kong.log.serialize()
-  local ok, err = timer_at(0, log, conf, message)
-  if not ok then
-    kong.log.err("failed to create timer: ", err)
+  local queue_id = get_queue_id(conf)
+  local q = queues[queue_id]
+  if not q then
+    local batch_max_size = conf.queue_size or 1
+    local process = function (entries)
+      return log(conf, entries)
+    end
+
+    local opts = {
+      retry_count    = conf.retry_count or 10,
```

Comment:
`retry_count` maybe `ngx.null`, because this field is not required, as well as `queue_size`, `flush_timeout`.

Reply:
Thanks for reminding this! I tested it and it is true that a field that is not mandatory can be set to `null` without any error.
So I made these fields `required = true` now, although I feel it is more like a nitpick because many of these fields in the OSS plugin do not have this limitation...